libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m

```diff
@@ -387,14 +368,139 @@ - (void)pgGetSoupSpec:(CDVInvokedUrlCommand *)command {
     } command:command];
 }
 
+- (void)pgGetAllGlobalStores:(CDVInvokedUrlCommand *)command {
+    [self runCommand:^(NSDictionary* argsDict) {
+        NSArray *allStoreNames = [SFSmartStore allGlobalStoreNames];
+        NSMutableArray *result = [NSMutableArray array];
+        if (allStoreNames.count >0 ) {
+            for(NSString *storeName in allStoreNames) {
+                NSMutableDictionary *dictionary = [NSMutableDictionary dictionary];
+                [dictionary setValue:storeName forKey:kStoreName];
+                [dictionary setValue:@YES forKey:kIsGlobalStoreArg];
+                [result addObject:dictionary];
+            }
+        }
+        return [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsArray:result];
+     }
+     command:command];
+}
+
+- (void)pgGetAllStores:(CDVInvokedUrlCommand *)command {
+    [self runCommand:^(NSDictionary* argsDict) {
+        NSArray *allStoreNames = [SFSmartStore allStoreNames];
+        NSMutableArray *result = [NSMutableArray array];
+        if (allStoreNames.count >0 ) {
+            for(NSString *storeName in allStoreNames) {
+                NSMutableDictionary *dictionary = [NSMutableDictionary dictionary];
+                [dictionary setValue:storeName forKey:kStoreName];
+                [dictionary setValue:@YES forKey:kIsGlobalStoreArg];
+                [result addObject:dictionary];
+            }
+           
+        }
+        return [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsArray:result];
+    } command:command];
+
+}
+
+- (void)pgRemoveStore:(CDVInvokedUrlCommand *)command {
+    [self runCommand:^(NSDictionary* argsDict) {
+        BOOL isGlobal = [self isGlobal:argsDict];
```

Comment:
This is another area where, similar to the cursor stuff, I wonder if it would be valuable to extract the store name and global flag into a class or struct, to consolidate/encapsulate the helper methods around it, and clean up this class.  It's turning into a sort of functional spaghetti at this point.